src/lib/actions/RuleDescribeAction.ts

```diff
@@ -0,0 +1,94 @@
+
+	public constructor(display: Display, ruleFilterFactory: RuleFilterFactory) {
+		this.display = display;
+		this.ruleFilterFactory = ruleFilterFactory;
+	}
+
+	public async validateInputs(_inputs: Inputs): Promise<void> { // eslint-disable-line @typescript-eslint/no-unused-vars
+		// Currently there is nothing to validate
+		return Promise.resolve();
+	}
+
+	public async run(inputs: Inputs): Promise<AnyJson> {
+		const jsonEnabled: boolean = inputs.json as boolean;
+
+		const ruleFilters: RuleFilter[] = this.ruleFilterFactory.createRuleFilters(inputs);
+
+		// TODO: Inject RuleManager as a dependency to improve testability by removing coupling to runtime implementation
+		const ruleManager = await Controller.createRuleManager();
+
+		const rules: DescribeStyledRule[] = await this.styleRules(ruleManager.getRulesMatchingOnlyExplicitCriteria(ruleFilters));
+		if (rules.length === 0) {
+			// If we couldn't find any rules that fit the criteria, we'll let the user know. We'll use .warn() instead of .log()
+			// so it's immediately obvious.
+			this.display.displayWarning(getMessage(BundleName.Describe, 'output.noMatchingRules', [inputs.rulename as string]));
+		} else if (rules.length > 1) {
+			// If there was more than one matching rule, we'll let the user know, but we'll still output all the rules.
+			const msg = getMessage(BundleName.Describe, 'output.multipleMatchingRules', [rules.length.toString(), inputs.rulename as string]);
+			this.display.displayWarning(msg);
+			rules.forEach((rule, idx) => {
+				this.display.displayStyledHeader(`Rule #${idx + 1}`);
+				this.displayStyledRule(rule, jsonEnabled);
+			});
+		} else {
+			// If there's exactly one rule, we don't need to do anything special, and can just log the rule.
+			this.displayStyledRule(rules[0], jsonEnabled);
+		}
+		// We need to return something for when the --json flag is used, so we'll just return the list of rules.
+		return deepCopy(rules);
+	}
+
+	private async styleRules(rules: Rule[]): Promise<DescribeStyledRule[]> {
+		// Opting to use .getAllEngines() instead of .getEnabledEngines() so we don't have to futz with the engineOptions param.
+
+		// TODO: Inject this as a dependency (maybe a factory) to improve testability by removing coupling to runtime implementation
+		const allEngines = await Controller.getAllEngines();
+
+		const enabledEngineNames: Set<string> = new Set();
+		for (const engine of allEngines) {
+			if (await engine.isEnabled()) {
+				enabledEngineNames.add(engine.getName());
+			}
+		}
+		return rules.map(r => {
+			const styledRule: DescribeStyledRule = {
+				...r,
+				runWith: r.isDfa ? Dfa.id : Run.id,
+				isPilot: r.isPilot,
+				enabled: enabledEngineNames.has(r.engine)
+			};
+			// Strip any whitespace off of the description.
+			styledRule.description = styledRule.description.trim();
+			return styledRule;
+		});
+	}
+
+	private displayStyledRule(rule: DescribeStyledRule, jsonEnabled: boolean): void {
+		new Ux({jsonEnabled: jsonEnabled})
```

Comment:
If our goal is to modularize the logging, then we should have a TODO here for calling `styledObject` via some helper method on a `Display` instead of by directly instantiating the UX object.

Reply:
Oh I didn't notice this. Yes this old code should be moved to the new Display world. I'll simply add a TODO for now.

Reply:
Fixed - squashed into part 10 with rebase and then force push.